system/handlers/formcontrols/SiteDateFormatPicker.cfc

```diff
@@ -0,0 +1,33 @@
+/**
+ * @feature presideForms and sites
+ */
+component {
+
+	property name="i18n"              inject="i18n";
+
+	public string function index( event, rc, prc, args={} ) output=false {
+		var settings       = getSetting( "datetime.formats" );
+		var dateFormatType = args.dateFormatType ?: "";
+		var items          = settings[dateFormatType];
```

Comment:
This assumes we've passed a valid dateFormatType that exists as a struct key. And in the line above it is defaulting to an empty string.